include/xtypes/DynamicData.hpp

```diff
@@ -439,6 +650,214 @@ class WritableDynamicDataRef : public ReadableDynamicDataRef
         catch(bool value) { return value; }
     }
 
+    /// \brief Class used for iterate WritableDynamicDataRef
+    class Iterator : public ReadableDynamicDataRef::Iterator
+    {
+    public:
+        Iterator(
+                WritableDynamicDataRef& ref)
+            : ReadableDynamicDataRef::Iterator(ref, false)
+        {}
+
+        WritableDynamicDataRef operator * ()
+        {
+            return ReadableDynamicDataRef::Iterator::operator*();
+        }
+
+    protected:
+
+        friend class WritableDynamicDataRef;
+
+        Iterator(
+                WritableDynamicDataRef& ref,
+                bool end)
+            : ReadableDynamicDataRef::Iterator(ref, end)
+        {
+        }
+    };
+
+    /// \brief Returns the initial iterator of a collection dynamic data.
+    /// \pre The DynamicData must represent a CollectionType.
+    /// \returns The initial iterator.
+    Iterator begin()
+    {
+        assert(type_.is_collection_type());
+        return Iterator(*this);
+    }
+
+    /// \brief Returns the final iterator of a collection dynamic data.
+    /// \pre The DynamicData must represent a CollectionType.
+    /// \returns The final iterator.
+    Iterator end()
+    {
+        assert(type_.is_collection_type());
+        return Iterator(*this, true);
+    }
+
+    class MemberPair : public ReadableDynamicDataRef::MemberPair
+    {
+    public:
+        MemberPair(
+                const Member& member,
+                uint8_t* data)
+            : ReadableDynamicDataRef::MemberPair(member, data)
+        {}
+
+        WritableDynamicDataRef data()
+        {
+            return WritableDynamicDataRef(member_.type(), data_);
+        }
+
+        WritableDynamicDataRef data() const
+        {
+            return WritableDynamicDataRef(member_.type(), data_);
+        }
+
+        template<typename T, class = PrimitiveOrString<T>>
+        WritableDynamicDataRef& operator = (
+                const T& other)
+        {
+            return data().operator=(other);
+        }
+
+        WritableDynamicDataRef& operator = (
+                const std::string& other)
+        {
+            return data().operator=(other);
+        }
+
+        WritableDynamicDataRef& operator = (
+                const std::wstring& other)
+        {
+            return data().operator=(other);
+        }
+
+        ReadableDynamicDataRef cref() const
+        {
+            return data().cref();
+        }
+
+        template<typename T, class = PrimitiveOrString<T>>
+        const T& value()
```

Comment:
Why do not return the only the data, and the left the user top chose what to do with this data? 
From a programmer point of view, each time that DynamicData interface changes, you need to change interface this too.